public/coffee/ide/editor/directives/aceEditor/track-changes/TrackChangesManager.coffee

```diff
@@ -233,6 +239,14 @@ define [
 			else
 				throw new Error("unknown change: #{JSON.stringify(change)}")
 
+		bulkAcceptChangeIds: (change_ids) ->
+			@rangesTracker.removeChangeIds(change_ids)
+			@updateAnnotations()
+
+		bulkRejectChangeIds: (change_ids) ->
+			for change_id in change_ids
+				@rejectChangeId change_id 
```

Comment:
This is going to be bad O(N^2) behaviour too. Each rejectChangeId will scan the entire changes array (O(N) in changes length), and do a look up between ace lines and offsets (O(N) in document size). This might require quite a bit of a refactor :( Have you tested it with a lot of changes for hanging the browser etc?